internal/services/report_processor.go

```diff
@@ -117,44 +117,48 @@ func ProcessReport(msg *kafka.Message) {
 				continue
 			}
 
-			usage_data_byte, err := kruize.Update_results(experiment_name, k8s_object)
-			if err != nil {
-				log.Error(err)
-				continue
-			}
-
-			for _, data := range usage_data_byte {
+			k8s_object_chunks := utils.SliceK8sObjectToChunks(k8s_object)
 
-				interval_start_time, _ := utils.ConvertStringToTime(data.Interval_start_time)
-				interval_end_time, _ := utils.ConvertStringToTime(data.Interval_end_time)
-
-				if workload_metrics, err := model.GetWorkloadMetricsForTimestamp(experiment_name, interval_end_time); err != nil {
-					log.Errorf("Error while checking for workload_metrics record: %s", err)
-					continue
-				} else if !reflect.ValueOf(workload_metrics).IsZero() {
-					log.Debugf("workload_metrics table already has data for interval_end time: %v.", interval_end_time)
+			for _, chunk := range k8s_object_chunks {
+				usage_data_byte, err := kruize.Update_results(experiment_name, chunk)
+				if err != nil {
+					log.Error(err)
```

Comment:
Could you confirm what details present under `err`? Will it be sufficient for debugging?


If something went wrong/failure occurs for `update_results` api with certain chunk, there will be data loss. Am I correct? 
If this is so, is it expected behavior?

Reply:
Yes, along with unique identifier for resource, it is good to include chunk specific info. example -  time duration

Reply:
Yes @kgaikwad if something went wrong/failure occurs for update_results api with certain chunk, there will be data loss but we cannot help it.  example - if suppose kruize application is down then update_results api will fail and we cannot do anything here to stop further API requests.

Reply:
Yes are not dividing chunks based on time stamp, means it is not the case that every metrics in the chunk have the same time stamp. So I think timestamp cannot be the unique identifier.

Reply:
>  Yes @kgaikwad if something went wrong/failure occurs for update_results api with certain chunk, there will be data loss but we cannot help it. example - if suppose kruize application is down then update_results api will fail and we cannot do anything here to stop further API requests.

In my opinion, we should think on such scenarios and come with handling to minimize data loss. May be not for this PR but it is better to do brainstorming around.

> So I think timestamp cannot be the unique identifier.

Then could you please include experiment name and any field info that might be specific to that request? So it will be easy for identifying that this upload request not processed completely, data loss observed.

Reply:
I have added the `experiment_name` in the log and by default `request_id` will be getting logged with the every log message.

Reply:
For first part, could you add a backlog ticket so that it won't get ignored by time. Thanks!